runtime/src/comm/ofi/comm-ofi.c

```diff
@@ -2125,14 +2155,25 @@ void init_ofiEp(void) {
     avAttr.count *= numTxCtxs;
   }
 
-  OFI_CHK(fi_av_open(ofi_domain, &avAttr, &ofi_av, NULL));
+  // The EFA provider does not allow AVs to be shared among
+  // endpoints. For all other providers we'll share them to reduce
+  // overhead.
+
+  chpl_bool useSharedAv = !providerInUse(provType_efa);
```

Comment:
I added some comments to help clarify the control flow. I didn't make the change you suggested because you have to create the AV before the EP.

Reply:
Okay.  I  would have ordered the code a little differently, but I'll defer to author style.
I do think that here, as in the case further down, we should be using `DBG_AV` (newly created) instead of `DBG_TCIPS`. `TCIPS` is really for things related specifically to tx context info pointers, i.e., the `tciTab` and I see this as having more to do with address vectors.